map/autoware_lanelet2_map_validator/test/src/test_regulatory_elements_details_for_crosswalks.cpp

```diff
@@ -0,0 +1,190 @@
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include "lanelet2_map_validator/validators/crosswalk/regulatory_element_details_for_crosswalks.hpp"
+#include "map_validation_tester.hpp"
+
+#include <gtest/gtest.h>
+#include <lanelet2_core/LaneletMap.h>
+
+class TestRegulatoryElementsDetailsForCrosswalks : public MapValidationTester
+{
+private:
+};
+
+TEST_F(TestRegulatoryElementsDetailsForCrosswalks, ValidatorAvailability)  // NOLINT for gtest
+{
+  std::string expected_validator_name =
+    lanelet::validation::RegulatoryElementsDetailsForCrosswalksValidator::name();
+
+  lanelet::validation::Strings validators =
+    lanelet::validation::availabeChecks(expected_validator_name);  // cspell:disable-line
+
+  const uint32_t expected_validator_num = 1;
+  EXPECT_EQ(expected_validator_num, validators.size());
+  EXPECT_EQ(expected_validator_name, validators[0]);
+}
+
+TEST_F(TestRegulatoryElementsDetailsForCrosswalks, WrongRefersType)  // NOLINT for gtest
+{
+  load_target_map("crosswalk/crosswalk_with_wrong_refers_type.osm");
+
+  lanelet::validation::RegulatoryElementsDetailsForCrosswalksValidator checker;
+  const auto & issues = checker(*map);
+
+  EXPECT_EQ(issues.size(), 1);
+  EXPECT_EQ(issues[0].id, 18);
+  EXPECT_EQ(issues[0].severity, lanelet::validation::Severity::Error);
+  EXPECT_EQ(issues[0].primitive, lanelet::validation::Primitive::Lanelet);
+  EXPECT_EQ(
+    issues[0].message, "Refers of crosswalk regulatory element must have type of crosswalk.");
+}
+
+TEST_F(TestRegulatoryElementsDetailsForCrosswalks, WrongRefLineType)  // NOLINT for gtest
+{
+  load_target_map("crosswalk/crosswalk_with_wrong_ref_line_type.osm");
+
+  lanelet::validation::RegulatoryElementsDetailsForCrosswalksValidator checker;
+  const auto & issues = checker(*map);
+
+  EXPECT_EQ(issues.size(), 1);
+  EXPECT_EQ(issues[0].id, 27);
+  EXPECT_EQ(issues[0].severity, lanelet::validation::Severity::Error);
+  EXPECT_EQ(issues[0].primitive, lanelet::validation::Primitive::LineString);
+  EXPECT_EQ(
+    issues[0].message, "ref_line of crosswalk regulatory element must have type of stopline.");
+}
+
+TEST_F(TestRegulatoryElementsDetailsForCrosswalks, WrongPolygonType)  // NOLINT for gtest
+{
+  load_target_map("crosswalk/crosswalk_with_wrong_polygon_type.osm");
+
+  lanelet::validation::RegulatoryElementsDetailsForCrosswalksValidator checker;
+  const auto & issues = checker(*map);
+
+  EXPECT_EQ(issues.size(), 1);
+  EXPECT_EQ(issues[0].id, 24);
+  EXPECT_EQ(issues[0].severity, lanelet::validation::Severity::Error);
+  EXPECT_EQ(issues[0].primitive, lanelet::validation::Primitive::Polygon);
+  EXPECT_EQ(
+    issues[0].message,
+    "Crosswalk polygon of crosswalk regulatory element must have type of Crosswalk_polygon.");
```

Comment:
```suggestion
    "Crosswalk polygon of crosswalk regulatory element must have type of crosswalk_polygon.");
```

And fix [here](https://github.com/autowarefoundation/autoware_tools/blob/2e93ca14722ff6ea97adf47f636e8614b3529e4a/map/autoware_lanelet2_map_validator/src/validators/crosswalk/regulatory_element_details_for_crosswalks.cpp#L122), too.